Add Player and ScreenBoundaries

FT-Fumetsu opened the pull request:
Le joueur a été ajouté. Il peut se déplacer horizontalement et il est bloqué aux bordures de l'écran peu importe son sprite.

FT-Fumetsu committed:
Add Player and ScreenBoundaries

Le joueur a été ajouté. Il peut se déplacer horizontalement et il est bloqué aux bordures de l'écran peu importe son sprite.

DomePuma commented on Assets/05-Scripts/PlayerScripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,23 @@
+        
+        private float _moveInput;
+
+        private void Update()
```
2 choses :
- Pourquoi utiliser de la physique quand on peut juste déplacer l'objet sans
- Quand manipule de la physique on utilise le FixedUpdate

FT-Fumetsu replied:
Je vois pas comment le déplacer sans utiliser la physique

DomePuma replied:
Juste en déplaçant sa position

FT-Fumetsu replied:
Ok, j'ai trouvé avant mais merci 👍

DomePuma commented on Assets/05-Scripts/PlayerScripts/ScreenBoundaries.cs:
```diff
@@ -0,0 +1,43 @@
+
+        private void Start()
+        {
+            var spriteRenderer = GetComponent<SpriteRenderer>();
```
Pourquoi utiliser un var quand on est déjà sûr du type qu'on veut récupérer ?

FT-Fumetsu replied:
Alors, je ne vois pas comment faire pour ne pas utiliser le var, ou alors j'ai pas compris ce que je devais faire o_O

DomePuma replied:
Ton Type de variable (donc dans ton cas SpriteRenderer) tu le connais déjà et tu peux pas récupérer un autre type que celui là avec ta méthode, donc utiliser var est inutile car tu sais déjà ce que tu vas récupérer et juste forcer à la machine à devoir faire une vérification du type pour savoir c'est lequel

FT-Fumetsu committed:
Update scripts after Pull Request

J'ai modifié le PlayerMovement pour qu'il n'utilise pas la physique et j'ai enlevé le "var" pour éviter une double vérification (je sais pas si c'est ce qu'il fallait faire, mais je ne pouvais pas enlever le "var" et laisser comme ça et je trouvais pas ça logique de le remplacer par "SpriteRenderer")

DomePuma commented on Assets/05-Scripts/PlayerScripts/ScreenBoundaries.cs:
```diff
@@ -11,9 +11,8 @@ public class ScreenBoundaries : MonoBehaviour
-            var spriteRenderer = GetComponent<SpriteRenderer>();
-            _playerHalfWidth = spriteRenderer.bounds.extents.x;
-            _playerHalfHeight = spriteRenderer.bounds.extents.y;
+            _playerHalfWidth = GetComponent<SpriteRenderer>().bounds.extents.x;
```
C'est encore pire, juste tu type la variable et après tu l'utilises, pas besoin de faire 2x GetComponent
```
SpriteRenderer spriteRenderer = GetComponent<SpriteRenderer>();
_playerHalfWidth = spriteRenderer .bounds.extents.x;
_playerHalfWidth = spriteRenderer .bounds.extents.y;
```

FT-Fumetsu replied:
Ok, je le voyais pas comme ça mais ok

DomePuma reviewed (changes requested):
Aussi faut utiliser le design patern Command pour contrôler le joueur, Logan a envoyé un lien sur le serveur

FT-Fumetsu committed:
Add Command Pattern And Remove PlayerMovement.cs

DomePuma commented:
J'ai oublié de le mettre, mais dans InputHandler la variable moveSpeed manque l'underscore, pense à le rajouter

LahessAimere commented on Assets/00-Scenes/SampleScene.unity:
Changes le nom de la scène parce que SampleScene, on ne sait pas à quoi ça correspond

FT-Fumetsu commented:
Ah ok my bad, je le fais de suite

LahessAimere commented on Assets/00-Scenes/SampleScene.unity:
Et je n'ai pas compris, elle te sert à quoi la globale light 2D ?

FT-Fumetsu replied:
Je n'y ai pas touché, ça y était déjà dans la scène

FT-Fumetsu committed:
Nomenclature Update

LahessAimere reviewed (commented):
Tu m'avais demandé de l'aide pour les diagrammes UML Mais du coup j'ai juste pris l'exemple de command pattern comme tu l'utilises
![image](https://github.com/user-attachments/assets/a5876d5c-9aa7-4dbc-ba92-132158e09f5a)

mais du coup, je vois que ça ne correspond pas trop à ce que t'as fait
Parce que t'as pas d'émetteur (Invoke) ni de récepteur (Receive)
Tu vas devoir faire des changements dans tes scripts alors je sais pas si c'est grave il faut que tu demandes à Cédric, mais du coup si tu veux changer ça je te conseille de changer l'update dans InputHandler et de plutôt le mettre en public Command, demande d'abord à cédric si c'est grave ou pas et tu me diras

FT-Fumetsu committed:
Change Scene Name And Remove Light 2D

FT-Fumetsu commented:
J'ai envoyé un message à Cedric, j'attends sa réponse

FT-Fumetsu commented:
Comment ça je l'ai close o_O
